### ISSUE TITLE
Automate janusgraph-cfg.md generation

### ISSUE THREAD

**Issue by porunov** [JanusGraph/janusgraph#1816]
Currently users should execute `mvn --quiet clean install -DskipTests=true -pl janusgraph-doc -am` after they added a new configuration option into JanusGraph. If they don't execute this command they will get Travis CI failure like in #1732 (https://travis-ci.org/JanusGraph/janusgraph/jobs/591029552).
Users may not know about this and it leads to not understanding the problem (why Travis CI fails).
I propose to automatically execute `mvn --quiet clean install -DskipTests=true -pl janusgraph-doc -am` in Travis CI if it fails with error in `janusgraph-cfg.md`. In such case, users don't need to execute `mvn --quiet clean install -DskipTests=true -pl janusgraph-doc -am` and configuration references will be generated automatically when it needs to update configuration references.

**Comment by porunov** [comment]
@farodin91 What do you think? Can we automate it? I.e. check for the error type and then generate configurations if it is needed? If Travis CI fails second time then the problem isn't in `janusgraph-cfg.md`.

**Comment by porunov** [comment]
I would like Travis to not produce this error https://travis-ci.org/JanusGraph/janusgraph/jobs/591029552 .
Instead of that error I would like that Travis would execute `mvn --quiet clean install -DskipTests=true -pl janusgraph-doc -am` and then produced the documentation from the result.

**Comment by farodin91** [comment]
This is not so easy, because you have to execute `mvn --quiet clean install -DskipTests=true -pl janusgraph-doc -am` on all version we release as documentation.

**Comment by porunov** [comment]
This is true, on all versions but we will execute it only when configuration reference must be changes. In all other cases we will not execute `mvn --quiet clean install -DskipTests=true -pl janusgraph-doc -am`

**Comment by porunov** [comment]
This will be a better option then telling contributors to run `mvn --quiet clean install -DskipTests=true -pl janusgraph-doc -am` again and wait until all Travis tests pass again just because of documentation change.

**Comment by farodin91** [comment]
If we change anything in our build system than we have to do it for all previous version. So, we have to maintain old version for much longer time. We also loose that the documentation is fully generated and viewable on GitHub or locally without building anything.

**Comment by FlorianHockmann** [comment]
When contributors modify config options, wouldn't we expect them to run a full build on their systems anyway (e.g. with just `mvn clean install -DskipTests`)? That would also generate this Markdown file, wouldn't it?

I agree in general that the docs should be up to date in our repository.

**Comment by porunov** [comment]
> When contributors modify config options, wouldn't we expect them to run a full build on their systems anyway (e.g. with just mvn clean install -DskipTests)?

As practice shows, they wouldn't. Check the next PRs where we have (or had) those problems which I've described: #1732 #1804

**Comment by farodin91** [comment]
> When contributors modify config options, wouldn't we expect them to run a full build on their systems anyway (e.g. with just `mvn clean install -DskipTests`)? That would also generate this Markdown file, wouldn't it?

We could add this to the PR template.
